hassio/src/dialogs/snapshot/dialog-hassio-create-snapshot.ts

```diff
@@ -0,0 +1,365 @@
+import { compare } from "../../../../src/common/string/compare";
+import "../../../../src/components/buttons/ha-progress-button";
+import "../../../../src/components/ha-checkbox";
+import type { HaCheckbox } from "../../../../src/components/ha-checkbox";
+import { createCloseHeading } from "../../../../src/components/ha-dialog";
+import "../../../../src/components/ha-formfield";
+import "../../../../src/components/ha-radio";
+import type { HaRadio } from "../../../../src/components/ha-radio";
+import "../../../../src/components/ha-settings-row";
+import { extractApiErrorMessage } from "../../../../src/data/hassio/common";
+import {
+  createHassioFullSnapshot,
+  createHassioPartialSnapshot,
+  HassioFullSnapshotCreateParams,
+  HassioPartialSnapshotCreateParams,
+  HassioSnapshot,
+} from "../../../../src/data/hassio/snapshot";
+import { showAlertDialog } from "../../../../src/dialogs/generic/show-dialog-box";
+import { PolymerChangedEvent } from "../../../../src/polymer-types";
+import {
+  haStyle,
+  haStyleDialog,
+  haStyleScrollbar,
+} from "../../../../src/resources/styles";
+import { HomeAssistant } from "../../../../src/types";
+import { HassioCreateSnapshotDialogParams } from "./show-dialog-hassio-create-snapshot";
+
+interface CheckboxItem {
+  slug: string;
+  checked: boolean;
+  name?: string;
+  version?: string;
+}
+
+@customElement("dialog-hassio-create-snapshot")
+class HassioCreateSnapshotDialog extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @internalProperty() private _snapshotName = "";
+
+  @internalProperty() private _snapshotPassword = "";
+
+  @internalProperty() private _snapshotHasPassword = false;
+
+  @internalProperty() private _snapshotType: HassioSnapshot["type"] = "full";
+
+  @internalProperty() private _dialogParams?: HassioCreateSnapshotDialogParams;
+
+  @internalProperty() private _addonList: CheckboxItem[] = [];
+
+  @internalProperty() private _folderList: CheckboxItem[] = [
+    {
+      slug: "homeassistant",
+      checked: true,
+    },
+    { slug: "ssl", checked: true },
+    { slug: "share", checked: true },
+    { slug: "media", checked: true },
+    { slug: "addons/local", checked: true },
+  ];
+
+  @internalProperty() private _error = "";
+
+  public showDialog(params: HassioCreateSnapshotDialogParams) {
+    this._dialogParams = params;
+    this._addonList = this._dialogParams.supervisor.supervisor.addons
+      .map((addon) => ({
+        slug: addon.slug,
+        name: addon.name,
+        version: addon.version,
+        checked: true,
+      }))
+      .sort((a, b) => compare(a.name, b.name));
+  }
+
+  public closeDialog() {
+    this._dialogParams = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected render(): TemplateResult {
+    if (!this._dialogParams) {
+      return html``;
+    }
+    return html`
+      <ha-dialog
+        class="ha-scrollbar"
```

Comment:
Are we using this in dialogs now too? Bit weird to use it in just one?

Reply:
Nope, that did not work